### PR TITLE
Separate out migrations and add route to retrieve store info

### PR DESCRIPTION
**Pull request by wenhaogoh**

### CODE REVIEW

File: src/db/migrations/20201023172243-create-product.js [hunk 1]
```diff
@@ -6,45 +6,43 @@ module.exports = {
...
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
     });
   },
-
-  down: async queryInterface => {
+  down: async (queryInterface, _Sequelize) => {
```
Comment (zatkiller) [inline src/db/migrations/20201023172243-create-product.js]: Just checking why _Sequelize parameter needed to be added in?

File: src/controllers/storeController.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
-import Store from '../models/Store';
```
Comment (ivantjh) [inline src/controllers/storeController.ts:1]: We can just return the address in `/store/:id`, no point having another call separately for that.

Comment (ivantjh) [reply src/controllers/storeController.ts:1]: Ideally, all of the calls that get stores should return address inside also as it is a common property.

File: src/controllers/storeController.ts [hunk 3]
```diff
@@ -59,8 +61,44 @@ async function destroyStore(req: Request, res: Response, next: NextFunction) {
   }
 }
 
+async function retrieveHawkerCentre(req: Request, res: Response, next: NextFunction) {
```
Comment (ivantjh) [inline src/controllers/storeController.ts]: Can just `retrieveHawkerCentre` in `retrieveStore`.

Comment (ivantjh) [reply src/controllers/storeController.ts]: Should be able to retrieve address with store using `include`, e.g. 
`const store = await Store.findOne({ include: HawkerCentre, id: id  });`.

File: src/models/HawkerCentre.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      type: DataTypes.STRING,
+    },
+    regionId: {
+      type: DataTypes.INTEGER,
+    },
+  },
+  {sequelize}
+);
+
+HawkerCentre.hasMany(Store, {foreignKey: 'hawkerCentreId'});
```
Comment (ivantjh) [inline src/models/HawkerCentre.ts:34]: If possible, define the associations in pairs,
https://sequelize.org/master/manual/assocs.html#why-associations-are-defined-in-pairs-

**Review by ivantjh — CHANGES_REQUESTED** [review]

File: src/db/migrations/20201023172243-create-product.js (hunk 1 above)
Comment (wenhaogoh) [reply src/db/migrations/20201023172243-create-product.js]: There is no need. I will remove it, thanks!

File: src/controllers/storeController.ts (hunk 2 above)
Comment (wenhaogoh) [reply src/controllers/storeController.ts:1]: 👍

File: src/controllers/storeController.ts (hunk 3 above)
Comment (wenhaogoh) [reply src/controllers/storeController.ts]: 👍

File: src/models/HawkerCentre.ts (hunk 4 above)
Comment (wenhaogoh) [reply src/models/HawkerCentre.ts:34]: 👍

**Review by zatkiller — APPROVED** [review]
LGTM

File: src/models/HawkerCentre.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Store.belongsTo(HawkerCentre, {foreignKey: 'hawkerCentreId'});
```
Comment (ivantjh) [inline src/models/HawkerCentre.ts:35]: Declare in `Store` class haha

File: src/models/Region.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    name: {
+      type: DataTypes.STRING,
+    },
+  },
+  {sequelize}
+);
+
+Region.hasMany(HawkerCentre, {foreignKey: 'regionId'});
+HawkerCentre.belongsTo(Region, {foreignKey: 'regionId'});
```
Comment (ivantjh) [inline src/models/Region.ts:26]: Same for here, declare in respective classes

File: src/models/Store.ts [hunk 7]
```diff
@@ -18,15 +22,24 @@ Store.init(
...
+    },
+    hawkerCentreId: {
+      type: DataTypes.INTEGER,
+    },
   },
   {sequelize}
 );
 
 Store.hasMany(Product, {foreignKey: 'storeId'});
+Product.belongsTo(Store);
```
Comment (ivantjh) [inline src/models/Store.ts:43]: This should go to `Product`.

**Review by ivantjh — APPROVED** [review]
LGTM!

**Pull request merged**